### PR TITLE
refactor: use instance field for cache

### LINKED ISSUE #87: Unnecessary operations on each request
On every request received by artipie (when using the default Pie slice), we do the following operations:
* retrieve storage from settings, which reads the yaml content (does not read the file again, but the file content as read at the first time) for "meta -> storage" entry.
* get the value of this storage scalar.

This is corresponding to that code:
```java
CompletableFuture.supplyAsync(
    () -> new Unchecked<>(this.settings::storage).value()
)
```

These two operations are redundant because they will always have the same result, there is no need to make it at every received request.
Ideally, if we won't support hot changing repos config files, we could implement a sort of cache that will return the whole slice (that is decorated by the AsyncSlice) based on the repo. Every repo is associated to a single slice, there is no need to recalculate that at every request. By "the whole slice that is decorated by the AsyncSlice", I meant this part of the code:
```java
CompletableFuture.supplyAsync(
    () -> new Unchecked<>(this.settings::storage).value()
)
.thenComposeAsync(
    storage -> storage.value(new Key.From(String.format("%s.yaml", repo)))
)
.thenApply(content -> new RepoConfig(content))
.thenCompose(Pie::sliceForConfig)

```

### PR DESCRIPTION
**Pull request by genryxy**
Closes #87
Used instance field instead of static because this instance is created only once. I didn't refactor static field in `RepositoriesFromStorage` because instances of this class are created many times

### CODE REVIEW

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks

**Comment by genryxy** [comment]
@g4s8 could you please check/merge

**Comment by genryxy** [comment]
@g4s8 could you please check/merge

File: src/test/java/com/artipie/cache/CachedUsersTest.java [hunk 1]
```diff
@@ -67,14 +85,18 @@ void failsToGetUserWhenCacheContainsAnotherUser() {
...
             user, "", (usr, pass) -> Optional.of(new Authentication.User(user))
         );
         MatcherAssert.assertThat(
             target.user(user, "", (usr, pass) -> Optional.empty()).isPresent(),
             new IsEqual<>(true)
         );
+        MatcherAssert.assertThat(
+            "Size of cache differed from 1",
+            this.cache.size(),
+            new IsEqual<>(1L)
```
Comment (g4s8) [inline src/test/java/com/artipie/cache/CachedUsersTest.java:99]: It's a minor issue, but you can use [Matchers.hasSize](http://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/Matchers.html#hasSize-int-) next time.

**Review by g4s8 — APPROVED** [review]

**Pull request merged**